source/extensions/filters/http/health_check/health_check.cc

```diff
@@ -38,6 +38,14 @@ Http::FilterHeadersStatus HealthCheckFilter::decodeHeaders(Http::HeaderMap& head
     health_check_request_ = true;
     callbacks_->requestInfo().healthCheck(true);
 
+    // Hint to tracer to ignore the span and any subsequent child spans.
+    // Health check probes are providing an operational check to ensure that the
+    // service is performing correctly. The benefit of distributed tracing is to
+    // record the path (and latency) a business transaction takes through multiple services to
+    // help diagnose issues. Recording trace instances for health check provides
+    // limited value, but consumes resources in the tracing system.
+    callbacks_->activeSpan().setSampled(false);
```

Comment:
Sorry, I get why we don't want to trace health checks. I don't fully understand why disabling sampling makes the traces go away?

Reply:
Ah I see. Sorry, this is quite confusing. `setSampled()` to me says whether to sample or not, and not sampled means always send. Can you rename the method at the Envoy API layer or add more comments?

Reply:
I would prefer to leave the method name as it is consistent with the propagation property name, e.g. https://github.com/openzipkin/b3-propagation#sampling-decision

Reply:
OK then can you update the comment here? https://github.com/envoyproxy/envoy/pull/3578/files#diff-7b5f0d3c634c8c1ac2d4fcaa26541dd9R102

Reply:
Hope this next version is ok - happy for any suggestions on how to improve though.